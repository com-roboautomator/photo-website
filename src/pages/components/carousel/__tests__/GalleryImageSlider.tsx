import {fireEvent, render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import GalleryImageSlider from '../GalleryImageSlider'
import collection from 'src/assets/data/ImageDataStructure'

const coll: collection[] = [
    {
        key: 'first-test-collection-key',
        title: 'first-test-collection-title',
        index: 0,

        images: [
            {
                key: 'test-image-key',
                url: 'test-image-url',
                title: 'test-image-title',
                index: 0,
            },
        ],
    },
    {
        key: 'second-test-collection-key',
        title: 'second-test-collection-title',
        index: 1,

        images: [
            {
                key: 'test-image-key',
                url: 'test-image-url',
                title: 'test-image-title',
                index: 0,
            },
        ],
    },
    {
        key: 'third-test-collection-key',
        title: 'third-test-collection-title',
        index: 2,

        images: [
            {
                key: 'test-image-key',
                url: 'test-image-url',
                title: 'test-image-title',
                index: 0,
            },
        ],
    },
]

describe('Gallery Image Slider', () => {
    it('should match snapshot', () => {
        const ImageSliderRender = renderer
            .create(<GalleryImageSlider data={coll} />)
            .toJSON()
        expect(ImageSliderRender).toMatchSnapshot()
    })

    it('should render correctly', () => {
        const {getByTestId} = render(<GalleryImageSlider data={coll} />)

        expect(getByTestId('Image-Slider-Title')).toBeInTheDocument()
        expect(getByTestId('Buttom-Wrapper')).toBeInTheDocument()
        expect(getByTestId('ImageSlider-Card-Slider')).toBeInTheDocument()
        expect(
            getByTestId('Collection-Card-first-test-collection-title')
        ).toBeInTheDocument()
    })

    it('should increment image', () => {
        const {getByTestId} = render(
            <GalleryImageSlider data={coll} startingIndex={0} />
        )

        const firstImage = getByTestId(
            'Collection-Card-first-test-collection-title'
        )
        const secondImage = getByTestId(
            'Collection-Card-second-test-collection-title'
        )
        const thirdImage = getByTestId(
            'Collection-Card-third-test-collection-title'
        )

        expect(firstImage).toHaveStyle({opacity: '100%', transform: 'scale(1)'})
        expect(secondImage).toHaveStyle({
            opacity: '97%',
            transform: 'scale(0.8)',
            filter: 'blur(1px)',
        })
        expect(thirdImage).toHaveStyle({
            transform: 'scale(0.5)',
            opacity: '60%',
            filter: 'blur(4px)',
        })

        fireEvent.click(getByTestId('Arrow-Button-Right'))

        expect(firstImage).toHaveStyle({
            opacity: '97%',
            transform: 'scale(0.8)',
            filter: 'blur(1px)',
        })
        expect(secondImage).toHaveStyle({
            opacity: '100%',
            transform: 'scale(1)',
        })
        expect(thirdImage).toHaveStyle({
            opacity: '97%',
            transform: 'scale(0.8)',
            filter: 'blur(1px)',
        })
    })

    it('should decrement image', () => {
        const {getByTestId} = render(
            <GalleryImageSlider data={coll} startingIndex={2} />
        )

        const firstImage = getByTestId(
            'Collection-Card-first-test-collection-title'
        )
        const secondImage = getByTestId(
            'Collection-Card-second-test-collection-title'
        )
        const thirdImage = getByTestId(
            'Collection-Card-third-test-collection-title'
        )

        expect(thirdImage).toHaveStyle({opacity: '100%', transform: 'scale(1)'})
        expect(secondImage).toHaveStyle({
            opacity: '97%',
            transform: 'scale(0.8)',
            filter: 'blur(1px)',
        })
        expect(firstImage).toHaveStyle({
            transform: 'scale(0.5)',
            opacity: '60%',
            filter: 'blur(4px)',
        })

        fireEvent.click(getByTestId('Arrow-Button-Left'))

        expect(firstImage).toHaveStyle({
            opacity: '97%',
            transform: 'scale(0.8)',
            filter: 'blur(1px)',
        })
        expect(secondImage).toHaveStyle({
            opacity: '100%',
            transform: 'scale(1)',
        })
        expect(thirdImage).toHaveStyle({
            opacity: '97%',
            transform: 'scale(0.8)',
            filter: 'blur(1px)',
        })
    })

    it('should move on wheel scroll', () => {
        const {getByTestId} = render(
            <GalleryImageSlider data={coll} startingIndex={2} />
        )

        const firstImage = getByTestId(
            'Collection-Card-first-test-collection-title'
        )
        const secondImage = getByTestId(
            'Collection-Card-second-test-collection-title'
        )
        const thirdImage = getByTestId(
            'Collection-Card-third-test-collection-title'
        )

        expect(thirdImage).toHaveStyle({opacity: '100%', transform: 'scale(1)'})
        expect(secondImage).toHaveStyle({
            opacity: '97%',
            transform: 'scale(0.8)',
            filter: 'blur(1px)',
        })
        expect(firstImage).toHaveStyle({
            transform: 'scale(0.5)',
            opacity: '60%',
            filter: 'blur(4px)',
        })

        fireEvent.wheel(getByTestId('ImageSlider-Card-Slider'))

        expect(firstImage).toHaveStyle({
            opacity: '97%',
            transform: 'scale(0.8)',
            filter: 'blur(1px)',
        })
        expect(secondImage).toHaveStyle({
            opacity: '100%',
            transform: 'scale(1)',
        })
        expect(thirdImage).toHaveStyle({
            opacity: '97%',
            transform: 'scale(0.8)',
            filter: 'blur(1px)',
        })
    })
})
