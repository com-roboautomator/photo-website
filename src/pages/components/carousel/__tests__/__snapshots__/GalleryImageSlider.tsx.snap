// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery Image Slider should match snapshot 1`] = `
<main
  className="GalleryImageSlider-container-1"
>
  <div
    className="ImageSliderTitle-container-5"
    data-testid="Image-Slider-Title"
    style={
      Object {
        "borderColor": "",
      }
    }
  >
    
  </div>
  <div
    className="GalleryImageSlider-wrapper-2"
    data-testid="ImageSlider-Container"
  >
    <div
      className="GalleryImageSliderButtonWrapper-button_wrapper-6"
      data-testid="Buttom-Wrapper"
    >
      <div
        className="GalleryImageSliderButtonWrapper-arrow-7"
      >
        <div
          className="Arrow-container-8"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text Mui-disabled Mui-disabled"
            data-testid="Arrow-Button-Left"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="nata"
                className="Arrow-left-9"
                data-testid="Arrow-Button-Image"
                src="arrow-disabled.png"
              />
            </span>
          </button>
        </div>
      </div>
      <div
        className="GalleryImageSliderButtonWrapper-arrow-7"
      >
        <div
          className="Arrow-container-8"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text"
            data-testid="Arrow-Button-Right"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <img
                alt="nata"
                className="Arrow-right-10"
                data-testid="Arrow-Button-Image"
                src="arrow-enabled.png"
              />
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      className="GalleryImageSlider-card_slider-3"
      data-testid="ImageSlider-Card-Slider"
      onWheel={[Function]}
      style={
        Object {
          "height": "undefinedpx",
        }
      }
    >
      <div
        className="GalleryImageSlider-card_slider_wrapper-4"
        style={
          Object {
            "transform": "translateX(-0%)",
          }
        }
      >
        <div
          className="GalleryCollectionCard-spacing_selected-11"
          data-testid="Collection-Card-first-test-collection-title"
        >
          <div
            className="GalleryCollectionCard-container-15"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              data-testid="Collection-Card-Link"
              href="gallery/first-test-collection-key"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <div
                className="GalleryCollectionCard-image_wrapper-16"
              >
                <img
                  alt=""
                  data-testid="Collection-Card-Image"
                  height={180}
                  src="test-image-url"
                  width="350"
                />
                <div
                  className="GalleryCollectionCard-container-17"
                  data-testid="Gallery-Collection-Card-Text"
                >
                  <div
                    className="GalleryCollectionCard-tag-18"
                    data-testid="Gallery-Collection-Card-Tag"
                    style={
                      Object {
                        "borderBottomColor": undefined,
                      }
                    }
                  />
                  <div
                    className="GalleryCollectionCard-text-19"
                    data-testid="Gallery-Collection-Card-Content-Text"
                  >
                    first-test-collection-title
                    <div
                      className="GalleryCollectionCardItemsText-container-20"
                      data-testid="Gallery-Collection-Card-Items-Text"
                    >
                      1
                       Item
                    </div>
                  </div>
                  <div
                    className="GalleryCollectionCardViewIcon-container-21"
                  >
                    <div
                      className="GalleryCollectionCardViewIcon-image-22"
                    >
                      <img
                        alt="none"
                        data-testid="Gallery-Collection-Card-View-Icon-Image"
                        height="20"
                        src="collection-image-w.png"
                        width="20"
                      />
                    </div>
                    <div
                      className="GalleryCollectionCardViewIcon-text-23"
                      data-testid="Gallery-Collection-Card-View-Icon-Text"
                    >
                      VIEW
                    </div>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="GalleryCollectionCard-spacing_immediate_background-12"
          data-testid="Collection-Card-second-test-collection-title"
        >
          <div
            className="GalleryCollectionCard-container-15"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              data-testid="Collection-Card-Link"
              href="gallery/second-test-collection-key"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <div
                className="GalleryCollectionCard-image_wrapper-16"
              >
                <img
                  alt=""
                  data-testid="Collection-Card-Image"
                  height={180}
                  src="test-image-url"
                  width="350"
                />
                <div
                  className="GalleryCollectionCard-container-17"
                  data-testid="Gallery-Collection-Card-Text"
                >
                  <div
                    className="GalleryCollectionCard-tag-18"
                    data-testid="Gallery-Collection-Card-Tag"
                    style={
                      Object {
                        "borderBottomColor": undefined,
                      }
                    }
                  />
                  <div
                    className="GalleryCollectionCard-text-19"
                    data-testid="Gallery-Collection-Card-Content-Text"
                  >
                    second-test-collection-title
                    <div
                      className="GalleryCollectionCardItemsText-container-20"
                      data-testid="Gallery-Collection-Card-Items-Text"
                    >
                      1
                       Item
                    </div>
                  </div>
                  <div
                    className="GalleryCollectionCardViewIcon-container-21"
                  >
                    <div
                      className="GalleryCollectionCardViewIcon-image-22"
                    >
                      <img
                        alt="none"
                        data-testid="Gallery-Collection-Card-View-Icon-Image"
                        height="20"
                        src="collection-image-w.png"
                        width="20"
                      />
                    </div>
                    <div
                      className="GalleryCollectionCardViewIcon-text-23"
                      data-testid="Gallery-Collection-Card-View-Icon-Text"
                    >
                      VIEW
                    </div>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
        <div
          className="GalleryCollectionCard-spacing_distant_background-13"
          data-testid="Collection-Card-third-test-collection-title"
        >
          <div
            className="GalleryCollectionCard-container-15"
          >
            <a
              className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
              data-testid="Collection-Card-Link"
              href="gallery/third-test-collection-key"
              onBlur={[Function]}
              onFocus={[Function]}
            >
              <div
                className="GalleryCollectionCard-image_wrapper-16"
              >
                <img
                  alt=""
                  data-testid="Collection-Card-Image"
                  height={180}
                  src="test-image-url"
                  width="350"
                />
                <div
                  className="GalleryCollectionCard-container-17"
                  data-testid="Gallery-Collection-Card-Text"
                >
                  <div
                    className="GalleryCollectionCard-tag-18"
                    data-testid="Gallery-Collection-Card-Tag"
                    style={
                      Object {
                        "borderBottomColor": undefined,
                      }
                    }
                  />
                  <div
                    className="GalleryCollectionCard-text-19"
                    data-testid="Gallery-Collection-Card-Content-Text"
                  >
                    third-test-collection-title
                    <div
                      className="GalleryCollectionCardItemsText-container-20"
                      data-testid="Gallery-Collection-Card-Items-Text"
                    >
                      1
                       Item
                    </div>
                  </div>
                  <div
                    className="GalleryCollectionCardViewIcon-container-21"
                  >
                    <div
                      className="GalleryCollectionCardViewIcon-image-22"
                    >
                      <img
                        alt="none"
                        data-testid="Gallery-Collection-Card-View-Icon-Image"
                        height="20"
                        src="collection-image-w.png"
                        width="20"
                      />
                    </div>
                    <div
                      className="GalleryCollectionCardViewIcon-text-23"
                      data-testid="Gallery-Collection-Card-View-Icon-Text"
                    >
                      VIEW
                    </div>
                  </div>
                </div>
              </div>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</main>
`;
